lib/fakefs/file.rb

```diff
@@ -222,6 +222,8 @@ def self.foreach(path, *args, &block)
     end
 
     def self.rename(source, dest)
+      return 0 if source == dest
```

Comment:
should go above the else so it still fails when the file does not exist

Reply:
actually it should go below the `if (target = FileSystem.find(source))`

Reply:
No problem, that should be us - updated the PR